controllers/helloworld_controller.go

```diff
@@ -0,0 +1,201 @@
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the Helloworld object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.13.0/pkg/reconcile
+func (r *HelloworldReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	_ = log.FromContext(ctx)
+
+	Logging(req, "Starting reconcile")
+
+	helloworld := &v1alpha1.Helloworld{}
+
+	if err := r.Get(ctx, req.NamespacedName, helloworld); err != nil {
+		if errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		log.Log.Error(err, "unable to fetch Helloworld")
+		return ctrl.Result{}, err
+	}
+
+	svc := &v1.Service{}
+	svcNamespacedName := types.NamespacedName{Namespace: helloworld.Namespace, Name: helloworld.Name + "-service"}
+
+	if err := r.Get(ctx, svcNamespacedName, svc); err != nil {
+		if errors.IsNotFound(err) {
+			Logging(req, "Creating a service")
+			r.Create(ctx, r.svcForHelloworld(helloworld))
+			return ctrl.Result{Requeue: true}, nil
+		}
+		log.Log.Error(err, "unable to fetch "+helloworld.Name+"'s service")
+		return ctrl.Result{}, err
+	}
+
+	Logging(req, "Service exist - "+svc.Name)
+
+	requiredReplicas := helloworld.Spec.ReplicaCount
+
+	Logging(req, "Required count of pods is "+fmt.Sprint(requiredReplicas))
+
+	podList := &v1.PodList{}
+	opts := []client.ListOption{
+		client.InNamespace(helloworld.Namespace),
+		client.MatchingLabels{"hw_name": helloworld.Name},
+	}
+
+	if err := r.List(ctx, podList, opts...); err != nil {
+		log.Log.Error(err, "unable to list "+helloworld.Name+"'s pods")
+		return ctrl.Result{}, err
+	}
+
+	currentReplicas := len(podList.Items)
+	Logging(req, "Current count of pods is "+fmt.Sprint(currentReplicas))
+
+	if requiredReplicas > int32(currentReplicas) {
+		for i := int32(0); i < requiredReplicas-int32(currentReplicas); i++ {
+			Logging(req, "Creating a pod number "+fmt.Sprint(i))
+			r.Create(ctx, r.podForHelloworld(helloworld))
+		}
+
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	countReady := 0
+	for _, pod := range podList.Items {
+		conditions := pod.Status.Conditions
+		for _, condition := range conditions {
+			if condition.Type == "Ready" && condition.Status == "True" {
+				countReady++
+			}
+		}
+	}
+
+	helloworld.Status.AvailableReplicas = int32(currentReplicas)
+	helloworld.Status.ReadyReplicas = int32(countReady)
+
+	Logging(req, "Updating status")
+
+	if err := r.Status().Update(ctx, helloworld); err != nil {
+		log.Log.Error(err, "unable to update Helloworld")
+		return ctrl.Result{}, err
+	}
+
+	Logging(req, "Finished succesfully")
+
+	return ctrl.Result{Requeue: false}, nil
+}
+
+func (r *HelloworldReconciler) svcForHelloworld(h *v1alpha1.Helloworld) *v1.Service {
```

Comment:
"h" is a bad param/var name.
a one-letter "var" is only acceptable (in our conventions..) as you used it in "(r * HelloworldReconciler)".